Add documentation on first party tracking setup (#645)
snowplow/documentation: docs/getting-started-on-snowplow-bdp-cloud/first-party-tracking/index.md

---
title: "First party tracking"
sidebar_position: 3
sidebar_custom_props:
  offerings:
    - cloud
---

# First party tracking

This feature lets you configure a custom collector domain to match the domain you wish to track. This allows for first-party cookies to be set on the server.
Enter the domain you wish to track, **any sub domain or multi level sub domain will be removed** and a cookie domain will be generated. If you wish to restrict the tracking to a sub domain the cookie domain can still be edited.

![enter_domain](images/Screenshot_enter_domain.png)

## DNS records

> The required DNS records will now be created, this may take several minutes.
> When the records are ready you will receive email confirmation.

Reviewer: ```suggestion
BDP Cloud will now generate the required DNS records. This may take several minutes.
When the records are ready, you will receive a confirmation by email.
```
Which email? The one used to sign up?

![create_dns_records](images/Screenshot_create_dns_records.png)

Once the DNS records are available, copy them into your domain provider.

> We will attempt to verify that the records have been correctly applied up to 72 hours after they have been generated.

Reviewer: What does this mean? The first time I read it I interpreted it as “at a single arbitrary point between 0 and 72 hours after generating the records, we will attempt to verify them”. But now I think you probably want to say that we will be continuously trying to verify within the next 72 hours?

> We will attempt to verify that the records have been correctly applied up to 72 hours after they have been generated.
> 
> **After 72 hours** if the records are not verified the process will be terminated in a failed state and you will need to start over.

Reviewer: ```suggestion
The set of DNS records will contain a special record that allows BDP Cloud to verify that the setup is correct.

:::info

You will have to conclude this step within 72 hours. If BDP Cloud is unable to verify the setup within that timeframe, you will have to restart the process from the beginning.

:::
```

Edited for clarity, used more active language (BDP Cloud vs passive voice), added more emphasis on the 72 hours

> When verification has been completed the status in the bottom left will change and you will receive email confirmation.

Reviewer: ```suggestion
Once BDP Cloud verifies your DNS setup, the status in the bottom left will change and you will receive a confirmation by email.
```

![dns_records](images/Screenshot_dns_records.png)

> ## Implement new tracking snippet

Reviewer: ```suggestion
## Update your tracking code
```

> In console navigate to “**Start tracking events**”, here you will find the JavaScript tracking snippet.

Reviewer: Provide the context first:
* to use first party tracking, users need to point their tracking code to the new URL (which is the cookie domain?)
* users who have previously deployed the tracking snippet provided in the UI can grab a new one in the UI
* users who implemented some other tracker (e.g. mobile) need to make this change manually

Then you can add the specific steps for the UI flow.

The code snippet will show the default collector URL, once the new custom domain is verified you can change the URL via the dropdown menu. 

> You now need to update the tracking snippet in your code base with the new URL to complete the implementation of First-party tracking.

Reviewer: > First-party

Could we make this uniform please?
* The UI says `first-party` (with a hyphen)
* The docs say `first party` (without a hyphen)
* This sentence uses a hyphen and a capital `F`
